crypto/threads_none.c

```diff
@@ -25,19 +25,22 @@ CRYPTO_RWLOCK *CRYPTO_THREAD_lock_new(void)
 
 int CRYPTO_THREAD_read_lock(CRYPTO_RWLOCK *lock)
 {
-    OPENSSL_assert(*(unsigned int *)lock == 1);
+    if (!ossl_assert(*(unsigned int *)lock == 1))
+        return 0;
```

Comment:
*Any* unexpected failure (assertion or otherwise) *could* be the result of memory corruption. Memory corruption is a kind of wild card possibility that you can't possibly sensibly try and detect. If you *really* have a situation where the *only* possible explanation for a failure is a memory corruption then I'd suggest that we shouldn't even bother asserting for it. Clearly in that case the code is so trivially obviously correct that adding an assert is kind of pointless. We *could* litter our code with hundreds of trivial assertions attempting to detect memory corruption - but I think that would be counter productive.